### PR TITLE
latest adpation of wrapper

### PR DESCRIPTION
**Pull request by ziquanyu**
there are two new folders with adapted changes according to vidhacne

### CODE REVIEW

File: nexus7/Application.mk [hunk 1]
```diff
@@ -0,0 +1,2 @@
+'APP_PLATFORM := android-21'
```
Comment (emilwestergren) [inline nexus7/Application.mk]: Remove: '

File: hry/VidhanceProcessor.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	SensorReader::getInstance()->disable();
+	vidhance_context_videoStop(context);
+}
+void VidhanceProcessor::processVideoCapture(frame_data_t frame) {
+	sp<GraphicBuffer> input = frame.inputBuffer;
+	sp<GraphicBuffer> output = frame.outputBuffer;
+	int horizontalStride = ALIGN(input->width, 1);
+	int verticalStride = ALIGN(input->height, 1);
+	int uvOffset = horizontalStride * verticalStride;
+//	// Image objects for Vidhance. Ownership is passed to Vidhance when processed, hence memory management is handled internally.
```
Comment (emilwestergren) [inline hry/VidhanceProcessor.cpp]: Remove commented code.

File: hry/VidhanceProcessor.cpp [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+			(void*)outputBuffer->getNativeBuffer(), (void*) outputBuffer->handle,
+			kean_math_intSize2D_new(outputBuffer->width * 4, 2 * outputBuffer->height / 3), horizontalStride, outputBuffer->format, uvOffset);
+
+	vidhance_sdk_vidhanceFrame framedata = vidhance_sdk_vidhanceFrame_new(kean_base_dateTime_new(frame.timestamp), kean_base_timeSpan_new(frame.lifetime),
+				kean_base_timeSpan_new(frame.rollingShutterSkew), (kean_draw_image*) inputImage, (kean_draw_image*) outputImage);
+
+	vidhance_context_processFrame(context, framedata);
+
+
+//	void *outputPtr;
```
Comment (emilwestergren) [inline hry/VidhanceProcessor.cpp]: Remove commented code.

File: nexus7/VidhanceProcessor.cpp [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+			for (int ind = 0; ind <(3 * input->height / 2); ind++) {
+				memcpy(inputBufferPtr, inputPtr, input->width);
+				inputPtr += (input->width);
+				inputBufferPtr += (rgbaStride*4);
+			}
+			memcpy(inputBufferPtr, inputPtr, yuvPadding);
+//			memcpy(inputBufferPtr, inputPtr, input->width * input->height * 1.5 + yuvPadding);
+			inputBuffer->unlock();
+
+//			inputPtr += (uvOffset - 15);
```
Comment (emilwestergren) [inline nexus7/VidhanceProcessor.cpp]: Remove commented code.

File: nexus7/VidhanceProcessor.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+#else
+#include "../HAL/HAL1/VideoProcessor.h"
+#endif
+namespace imint {
+class VidhanceProcessor: public VideoProcessor {
+public:
+	VidhanceProcessor(const char* cameraId);
+	virtual ~VidhanceProcessor();
+
+	//virtual void processPreview(sp<GraphicBuffer>, sp<GraphicBuffer>);
```
Comment (emilwestergren) [inline nexus7/VidhanceProcessor.h]: Remove commented code.

**Comment by davidhesselbom** [comment]
> vidhacne

Ouch.

**Pull request closed without merging**